lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
```

Comment:
Doesn't `args.first` just return the args were passed the first time the message was received, and ignore all the other times?  Or am I misunderstanding the structure of `args`?